src/details/ArborX_DetailsBoundingVolumeHierarchyImpl.hpp

```diff
@@ -490,7 +482,10 @@ BoundingVolumeHierarchyImpl<DeviceType>::queryDispatch(
 
   Kokkos::Profiling::popRegion();
 
-  if (max_results_per_query > buffer_size)
+  // The check for the number of results is for an unusual situation that may
+  // happen when either no results were found, or nothing was inserted inside a
+  // callback for found results.
+  if (n_results && (max_results_per_query > buffer_size))
```

Comment:
```suggestion
  if (n_results > 0 && (max_results_per_query > buffer_size))
```

Reply:
Meh, I don't think this is necessary.

Reply:
This is one of the things that make me think for a second what is meant. IMHO, it's totally worth to be explicit and not to rely on the implicit conversion to a bool value.